jena-arq/src/main/java/org/apache/jena/http/sys/ExecUpdateHTTPBuilder.java

```diff
@@ -200,28 +296,35 @@ private void ensureContext() {
             context = Context.create();
     }
 
-    private void ensureUpdateRequest() {
-        if ( updateOperations == null )
-            updateOperations = new UpdateRequest();
-    }
-
     public X build() {
         Objects.requireNonNull(serviceURL, "No service URL");
-        if ( updateOperations == null && updateString == null )
-            throw new QueryException("No update for UpdateExecutionHTTP");
-        if ( updateOperations != null && updateString != null )
-            throw new InternalErrorException("UpdateRequest and update string");
+        if ( updateEltAcc.isEmpty() )
+            throw new UpdateException("No update for UpdateExecutionHTTP");
```

Comment:
The original code used QueryException and I thought that it might have been a copy/paste issue. Then again, QueryParseException is also re-used for parse errors with update requests, so I am unsure what is the right choice her.